domain/controllernode/service/service.go

```diff
@@ -205,21 +205,31 @@ func (s *Service) IsControllerNode(ctx context.Context, nodeID string) (bool, er
 //
 // The following errors can be expected:
 // - [controllernodeerrors.NotFound] if the controller node does not exist.
-func (s *Service) SetAPIAddresses(ctx context.Context, controllerID string, addrs network.SpaceHostPorts, mgmtSpace network.SpaceInfo) error {
+func (s *Service) SetAPIAddresses(ctx context.Context, controllerID string, addrs network.SpaceHostPorts, mgmtSpace *network.SpaceInfo) error {
 	// We map the SpaceHostPorts addresses to controller api addresses by
 	// checking if the address is available for agents (this is the case if the
 	// space ID of the address matches the management space ID), and also by
 	// joining the address host and port to a string "host:port".
 	addresses := make([]controllernode.APIAddress, 0, len(addrs))
+	emptyAgentAddresses := true
 	for _, spHostPort := range addrs {
-		// Check if the address is available for agents.
-		isAvailableForAgents := spHostPort.SpaceID == mgmtSpace.ID
+		// Check if the address is available for agents. If no management space
+		// is set, all addresses are available for agents.
+		isAvailableForAgents := mgmtSpace == nil || spHostPort.SpaceID == mgmtSpace.ID
 		// Join the address host and port to a string "host:port".
 		address := net.JoinHostPort(spHostPort.Host(), strconv.Itoa(spHostPort.Port()))
 		addresses = append(addresses, controllernode.APIAddress{
 			Address: address,
 			IsAgent: isAvailableForAgents,
 		})
+		emptyAgentAddresses = emptyAgentAddresses && !isAvailableForAgents
+	}
+	// If we have filtered out all addresses, set all to agents to ensure that
+	// the API is always reachable for agents.
+	if emptyAgentAddresses {
```

Comment:
This is where we should log the warning.

Reply:
Sure, done.